src/metabase/models/params.clj

```diff
@@ -221,10 +211,19 @@
   `dashcards` must be hydrated with :card."
   [dashcards]
   (set/union
-   (set (lib.util.match/match (seq (dashcards->parameter-mapping-field-clauses dashcards))
-          [:field (id :guard integer?) _]
-          id))
-   (cards->card-param-field-ids (map :card dashcards))))
+    (set (for [{:keys [card] :as dashcard} dashcards
+               param    (:parameter_mappings dashcard)
+               :let     [field-clause (param-target->field-clause (:target param) (:card dashcard))]
+               :when    field-clause]
+           (or
+             ;; Get the field id from the field-clause if it contains it. This is the common case
+             ;; for mbql queries.
+             (lib.util.match/match-one field-clause [:field (id :guard integer?) _] id)
+             ;; Attempt to get the field clause from the model metadata corresponding to the field.
+             ;; This is the common case for native queries in which mappings from original columns
```

Comment:
We don't allow mapping between parameters and native models now. This is actually needed for mbql models because they have string-based refs and the above condition with `integer?` doesn't work. Maybe it's worth updating the comment